javarosa/src/main/java/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -59,10 +59,11 @@ public XPathFuncExpr() {
     } //for deserialization
 
     public XPathFuncExpr(XPathQName id, XPathExpression[] args) throws XPathSyntaxException {
-
-        if (id.name.equals("if")) {
-            if (args.length != 3) {
-                throw new XPathSyntaxException("if() function requires 3 arguments but " + args.length + " are present.");
+        if ("if".equals(id.name) && args.length != 3) {
+            throw new XPathSyntaxException("if() function requires 3 arguments but " + args.length + " are present.");
+        } else if ("cond".equals(id.name)) {
+            if (args.length < 3 || args.length % 2 != 1) {
+                throw new XPathSyntaxException("cond() function requires an odd number of arguments. " + args.length + " are present.");

```

Comment:
Technically someone could have tried to call `cond()` with 1 argument here, and then the error message would be confusing. Probably best to split up the 2 checks and have a different error message for `args.length < 3`, even though it's unlikely